src/lib/secrets_helper.ts

```diff
@@ -0,0 +1,191 @@
+
+import { Event as HandlerEvent } from '../function/event'
+
+const services = {
+  gitHub: null,
+  spotify: null,
+  salesforce: null,
+  stripe: null,
+}
+
+export type Service = {
+  friendlyServiceName: string
+  service: string
+  isLoggedIn: boolean
+  bearerToken: string | null
+  grantedScopes: Array<{
+    scope: string
+    scopeInfo: {
+      category: string | null
+      scope: string
+      display: string
+      isDefault: boolean
+      isRequired: boolean
+      description: string | null
+      title: string | null
+    }
+  }> | null
+}
+
+export type Services = typeof services
+
+export type ServiceKey = keyof Services
+
+export type ServiceTokens = Service
+
+export type NetlifySecrets = {
+  [K in ServiceKey]?: Service
+} & { [key: string]: Service }
+
+type OneGraphSecretsResponse = {
+  data?: {
+    me?: {
+      serviceMetadata?: {
+        loggedInServices: [Service]
+      }
+    }
+  }
+}
+
+const siteId = env.SITE_ID
+
+const camelize = function (text: string) {
+  const safe = text.replace(/[-_\s.]+(.)?/g, (_, sub) => (sub ? sub.toUpperCase() : ''))
+  return safe.slice(0, 1).toLowerCase() + safe.slice(1)
+}
+
+type ServiceNormalizeOverrides = {
+  GITHUB: string
+  [key: string]: string
+}
+
+// The services will be camelized versions of the OneGraph service enums
+// unless overridden by the serviceNormalizeOverrides object
+const serviceNormalizeOverrides: ServiceNormalizeOverrides = {
+  // Keys are the OneGraph service enums, values are the desired `secret.<service>` names
+  GITHUB: 'gitHub',
+}
+
+const oneGraphRequest = function (secretToken: string, requestBody: Uint8Array): Promise<OneGraphSecretsResponse> {
+  return new Promise((resolve, reject) => {
+    const port = 443
+
+    const options = {
+      host: 'serve.onegraph.com',
+      path: `/graphql?app_id=${siteId}`,
+      port,
+      method: 'POST',
+      headers: {
+        Authorization: `Bearer ${secretToken}`,
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        'Content-Length': requestBody ? Buffer.byteLength(requestBody) : 0,
+      },
+    }
+
+    const req = request(options, (res) => {
+      if (res.statusCode !== 200) {
+        return reject(new Error(String(res.statusCode)))
+      }
+
+      const body: Array<Uint8Array> = []
+
+      res.on('data', (chunk) => {
+        body.push(chunk)
+      })
+
+      res.on('end', () => {
+        const data = Buffer.concat(body).toString()
+        try {
+          const result: OneGraphSecretsResponse = JSON.parse(data)
+          resolve(result)
+        } catch (error) {
+          reject(error)
+        }
+      })
+    })
+
+    req.on('error', (error) => {
+      reject(error.message)
+    })
+
+    req.write(requestBody)
+
+    req.end()
+  })
+}
+
+const formatSecrets = (result: OneGraphSecretsResponse | undefined) => {
+  const responseServices = result?.data?.me?.serviceMetadata?.loggedInServices
+
+  if (responseServices) {
```

Comment:
I think this would be a bit easier to read if we reversed the condition of the if statement and did an early return, but this boils down to personal preference, so by all means feel free to ignore.